### PR TITLE
feat: Add withdrawn dataset toast (#1237)

### PR DESCRIPTION
**Pull request by leslieypark**
### Reviewers
**Functional:** 
@tihuan 

**Readability:** 
@seve

---

## Changes
- Add withdrawn dataset toast to collections page. When a user goes to an explorer link for a tombstoned dataset, the explorer will redirect the user back to the parent collections page with the URL query param of `tombstoned_dataset_id`.

Screen Shot 2021-09-08 at 10 02 10

## Definition of Done (from ticket)
Navigating to a tombstoned dataset result in a withdrawn dataset toast.

## QA steps (optional)
Run data portal locally and go to a collection. Add `?tombstoned_dataset_id=asdfasdf` to the URL and press enter. You should see the toast.

## Known Issues
The explorer side of this ticket (redirection) is being completed by @MDunitz.

### CODE REVIEW

File: frontend/src/views/Collection/index.tsx [hunk 1]
```diff
@@ -118,6 +121,16 @@ const Collection: FC = () => {
...
     collection.access_type === ACCESS_TYPE.WRITE && isPrivate;
 
+  const showTombstonedDatasetToast = () => {
+    Toast.show({
+      icon: IconNames.ISSUE,
+      intent: Intent.PRIMARY,
+      message:
+        `A dataset was withdrawn by ${collection.contact_name}. You've been redirected to the parent collection.`,
+    });
+    setHasTombestonedToastBeenShown(true)
```
Comment (leslieypark) [inline frontend/src/views/Collection/index.tsx:131]: It seems like this is what's causing this error:

```
Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.
```

Comment (leslieypark) [reply frontend/src/views/Collection/index.tsx:131]: Upon checking, it seems like the issue is coming from `Toast.show`. Without adding any other lines of code, I added the `Toast.show` snippet to the body of the component, and the error shows up.

**Comment by tihuan** [comment]
@leslieypark  This PR should fix the lint errors you have! https://github.com/chanzuckerberg/single-cell-data-portal/pull/1384

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks for adding this and all the comments too!

The linter fix is in this [PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/1384). Please feel free to merge it into this PR!

Thank you!

**Comment by tihuan** [comment]
Rerunning BE unit test to see if it's flaky 

Screen Shot 2021-09-08 at 1 02 03 PM

**Comment by leslieypark** [comment]
@tihuan Waiting for smoke test issue to be fixed before merging!

**Comment by tihuan** [comment]
Ah @leslieypark which smoke test needs to be fixed? So far I see all PR tests passing 😆 ?

**Pull request merged**